cpp/src/copying/copy.cu

```diff
@@ -38,18 +40,40 @@ struct copy_if_else_functor_impl {
   }
 };
 
+/**
+ * @brief Functor to fetch a device-view for the specified scalar/column_view.
+ */
+struct get_iterable_device_view {
+  template <typename T, CUDF_ENABLE_IF(std::is_same<T, cudf::column_view>::value)>
+  auto operator()(T const& input)
+  {
+    return cudf::column_device_view::create(input);
+  }
+
+  template <typename T, CUDF_ENABLE_IF(std::is_same<T, cudf::scalar>::value)>
+  auto operator()(T const& input)
+  {
+    return &input;
+  }
+};
+
 template <typename T>
 struct copy_if_else_functor_impl<T, std::enable_if_t<is_rep_layout_compatible<T>()>> {
   template <typename Left, typename Right, typename Filter>
-  std::unique_ptr<column> operator()(Left const& lhs,
-                                     Right const& rhs,
+  std::unique_ptr<column> operator()(Left const& lhs_h,
+                                     Right const& rhs_h,
                                      size_type size,
                                      bool left_nullable,
                                      bool right_nullable,
                                      Filter filter,
                                      rmm::cuda_stream_view stream,
                                      rmm::mr::device_memory_resource* mr)
   {
+    auto p_lhs      = get_iterable_device_view{}(lhs_h);
+    auto p_rhs      = get_iterable_device_view{}(rhs_h);
+    auto const& lhs = *p_lhs;
```

Comment:
If there isn't a nullptr check, why break this into multiple lines?

Reply:
`get_iterable_device_view()` returns a `unique_ptr<column_device_view>`. It needs to be kept alive, and can't be dereferenced directly.